commandapi-platforms/commandapi-bukkit/commandapi-bukkit-core/src/main/java/dev/jorel/commandapi/arguments/ListArgumentCommon.java

```diff
@@ -137,12 +138,18 @@ public <CommandSourceStack> List<T> parseArgument(CommandContext<CommandSourceSt
 		final String[] strArr = argument.split(Pattern.quote(delimiter));
 		final StringReader context = new StringReader(argument);
 		for (String str : strArr) {
-			if (!values.containsKey(str)) {
+			if (!allowAnyValue && !values.containsKey(str)) {
 				throw new SimpleCommandExceptionType(new LiteralMessage("Item is not allowed in list")).createWithContext(context);
 			} else if (!allowDuplicates && listKeys.contains(str)) {
 				throw new SimpleCommandExceptionType(new LiteralMessage("Duplicate arguments are not allowed")).createWithContext(context);
 			} else {
-				list.add(values.get(str));
+				if (allowAnyValue) {
+					// should only be used with string mappers so we can ignore the cast warning
+					list.add((T) str);
```

Comment:
```suggestion
					list.add(values.get(str));
```
As we have the map, I think we should use it instead of resorting to generic casts.

Reply:
@DerEchtePilz when I use the mapper it will give me a list of nulls. The only way I've found that works is to directly cast the string to T.

Reply:
Well, that sounds like a bug to me that should get fixed. I am currently not available to test anything related to the `ListArgument` but using the mapper should most definitely not result in `null` values.

Reply:
I'm not sure if it's a bug but maybe just a design flaw. The mapper value only works if the input is in the suggestions list.
That part might not be compatible with what we're trying to accomplish so that's why we made allowAnyValue only valid with strings since we need to cast the string input to T.